src/sfdc46/salesforceBulk/csv/csv_update_operator.bal

```diff
@@ -119,8 +119,30 @@ public type CsvUpdateOperator client object {
     # Get the results of the batch.
     #
     # + batchId - batch ID
+    # + numberOfTries - number of times checking the batch state
+    # + waitTime - time between two tries in ms
     # + return - Batch result as CSV if successful else SalesforceError occured
-    public remote function getBatchResults(string batchId) returns @tainted string | SalesforceError {
-        return self.httpBaseClient->getCsvRecord([JOB, self.job.id, BATCH, batchId, RESULT]);
+    public remote function getBatchResults(string batchId, int numberOfTries = 1, int waitTime = 3000) 
+        returns @tainted string | SalesforceError {
+        int counter = 0;
+        while (counter < numberOfTries) {
```

Comment:
I think we can extract out this method as a generic method as pass the getCsvRecord part as a function pointer. That way we get rid of the duplicate code. WDYT?

Reply:
Created an issue as an improvement.
https://github.com/wso2-ballerina/module-salesforce/issues/63